pkg/runutil/runutil.go

```diff
@@ -108,6 +115,12 @@ func CloseWithLogOnErr(logger log.Logger, closer io.Closer, format string, a ...
 	level.Warn(logger).Log("msg", "detected close error", "err", errors.Wrap(err, fmt.Sprintf(format, a...)))
 }
 
+// ExhaustCloseWithLogOnErr closes the io.ReadCloser with a log message on error but exhausts the reader before.
+func ExhaustCloseWithLogOnErr(logger log.Logger, r io.ReadCloser, format string, a ...interface{}) {
+	_, _ = io.Copy(ioutil.Discard, r)
```

Comment:
Technically log on error might mean checking err here as well (:  But I think it's fine for now.

Reply:
That's true but such an error message would be harmless and users would just ignore it, no?

Reply:
No if you want to exhaust and that failed, right?